code/datums/view.dm

```diff
@@ -0,0 +1,172 @@
+		assert_format()
+		return
+	update_pixel_format()
+
+///Resets the format type
+/datum/view_data/proc/assert_format()
+	winset(parent, "mapwindow.map", "zoom=0")
+
+///applies the current clients preferred pixel size setting
+/datum/view_data/proc/update_pixel_format()
+	winset(parent, "mapwindow.map", "zoom=[parent.prefs.pixel_size]")
+
+///applies the preferred clients scaling method
+/datum/view_data/proc/update_zoom_mode()
+	winset(parent, "mapwindow.map", "zoom-mode=[parent.prefs.scaling_method]")
+
+///Returns a boolean if the client has any form of zoom
+/datum/view_data/proc/is_zooming()
+	return (width || height)
+
+///Resets the zoom to the default string
+/datum/view_data/proc/reset_to_default()
+	width = 0
+	height = 0
+	apply()
+
+///adds the number inputted to the zoom and applies it
+/datum/view_data/proc/add(num_to_add)
+	width += num_to_add
+	height += num_to_add
+	apply()
+
+///adds the size, which can also be a string, to the default and applies it
+/datum/view_data/proc/add_size(toAdd)
+	var/list/new_size = getviewsize(toAdd)
+	width += new_size[1]
+	height += new_size[2]
+	apply()
+
+///sets the size, which can also be a string and applies it
+/datum/view_data/proc/set_view_radius_to(toAdd)
+	var/list/new_size = getviewsize(toAdd)  //Backward compatability to account
+	width = new_size[1] //for a change in how sizes get calculated. we used to include world.view in
+	height = new_size[2] //this, but it was jank, so I had to move it
+	apply()
+
+///sets width and height as numbers
+/datum/view_data/proc/set_width_and_height(new_width, new_height)
+	width = new_width
+	height = new_height
+	apply()
+
+///sets the width of the view
+/datum/view_data/proc/set_width(new_width)
+	width = new_width
+	apply()
+
+///sets the height of the view
+/datum/view_data/proc/set_height(new_height)
+	height = new_height
+	apply()
+
+///adds the inputted width to the view
+/datum/view_data/proc/add_to_width(width_to_add)
+	width += width_to_add
+	apply()
+
+///adds the inputted height to the view
+/datum/view_data/proc/add_to_height(height_to_add)
+	height += height_to_add
+	apply()
+
+///applies all current outstanding changes to the client
+/datum/view_data/proc/apply()
+	parent.change_view(get_client_view_size())
+	safe_apply_formatting()
+
+///supresses any further view changes until it is unsupressed
+/datum/view_data/proc/suppress()
+	suppress_changes = TRUE
+	apply()
+
+///unsupresses to allow further view changes
+/datum/view_data/proc/unsuppress()
+	suppress_changes = FALSE
+	apply()
+
+///returns the client view size in string format
+/datum/view_data/proc/get_client_view_size()
+	var/list/temp = getviewsize(default)
+	if(suppress_changes)
+		return "[temp[1]]x[temp[2]]"
+	return "[width + temp[1]]x[height + temp[2]]"
+
+///Zooms the client back in with an animate pretty simple
+/datum/view_data/proc/zoom_in()
+	reset_to_default()
+	animate(parent, pixel_x = 0, pixel_y = 0, 0, FALSE, LINEAR_EASING, ANIMATION_END_NOW)
```

Comment:
Just a useability comment, but if this is what's causing the smooth transition when leaving binoculars, please, for the love of God, make it a toggle or something. It's motion sickness inducing.

Reply:
it wasn't nevermind, thanks harry